approval_polls/static/approval_polls/create.js

```diff
@@ -9,17 +9,104 @@ $(function () {
 
     this.numChoiceFields++;
     formGroup = $("<div class='form-group'></div>");
-    input = $("<input class='form-control' type='text' maxlength=200 name='choice" +
+    input = $("<div class='input-group' id='div-choice" + this.numChoiceFields + "'><input class='form-control' type='text' maxlength=200 name='choice" +
             this.numChoiceFields + "' placeholder='Choice " +
-            this.numChoiceFields + "'>");
+            this.numChoiceFields + "'><span class='input-group-addon'>"+
+            "<a href='#' id='link-choice" + this.numChoiceFields + "' title='Add link' data-toggle='tooltip' data-placement='bottom'>"+
+            "<span class='glyphicon glyphicon-link'></span></a></span>" + 
+            "<input type='hidden' id='linkurl-choice" + this.numChoiceFields + "' name='linkurl-choice" + this.numChoiceFields + "' value=''></div>");
 
     formGroup.append(input);
 
     $('.form-group').last().after(formGroup);
+    $('[data-toggle=tooltip]').tooltip();
   };
 
   $('[data-toggle=tooltip]').tooltip();
   $('button#add-choice').click($.proxy(this.addChoiceField, this));
+  
+  /* Allow user to attach an external link to an option. */
+
+  // Event delegation to capture dynamically added links
+  $('.row-fluid').on('click', 'a', function() {
+    var alertDiv, alertDivId, currentUrl;
+    alertDivId = $(this).attr('id');
+    alertDivId = alertDivId.split('-').pop();
+
+    alertDiv = "<div class='alert alert-info' id='alert-" + alertDivId + "'>" + 
+      "<p><input type='text' class='form-control' id='url-" + alertDivId + "' placeholder='Link to URL'></p><p>" + 
+      "<button id='confirm-link-" + alertDivId + "' type='button' class='btn btn-success btn-xs'>Insert Link</button>" + 
+      " <button id='remove-link-" + alertDivId + "' type='button' class='btn btn-danger btn-xs'>Reset Link</button>" + 
+      " <button id='cancel-link-" + alertDivId + "' type='button' class='btn btn-info btn-xs'>Cancel</button></p></div>";
+
+    if ($('#alert-' + alertDivId).length == 0) {
```

Comment:
Using '===' instead of '=='  seems to be recommended for performance benefits when type conversion is not required. This [stackoverflow post](http://stackoverflow.com/questions/359494/does-it-matter-which-equals-operator-vs-i-use-in-javascript-comparisons) gives some good insight.